ambry-router/src/main/java/com/github/ambry/router/GetOperation.java

```diff
@@ -203,6 +203,8 @@ protected int getPrecedenceLevel(RouterErrorCode routerErrorCode) {
         return 7;
       case BlobDoesNotExist:
         return 8;
+      case TooManyRequests:
+        return 9;
```

Comment:
TooManyRequest should have precedence over other errors. If we have two errors, first one is BlobDoesNotExist, second one if TooManyRequest, then we should return TooManyRequest.